### PR TITLE
Added a user details page

### LINKED ISSUE #70: Display usernames differently
Usernames are now displayed as they are known at CNCZ. The bakers of a shift should now the first name of someone that orders a product, so they can shout it through the canteen if their tosti is ready. We should create a way in which people enter there first/last name and change the way it is displayed in the order views.

### LINKED ISSUE #71: Add a to_lowercase on science login name
science usernames are always lowercase, but OpenID is capital-sensitive. So it would be nice to make this conversion in our code

### PR DESCRIPTION
**Pull request by KiOui**
Added a page where users can see their user information.
Changed the OpenID functions and created a simple class.
Changed the OpenID request such that user information is automatically requested and stored. 

Closes #70 
Closes #71

### CODE REVIEW

**Comment by JobDoesburg** [comment]
Can't we just request the full name of the user via OpenID?

**Comment by JobDoesburg** [comment]
If not I think we should build some check on the lastname and the user name to have some fraud check. Or at least show names as `Firstname Lastname (username)`, e.g. "John Doe (jdoe)" so the username remains visible and we do have some unique part in all displayed names

**Comment by KiOui** [comment]
> Can't we just request the full name of the user via OpenID?

I don't think this is possible

**Comment by JobDoesburg** [comment]
It is, by adding `"openid.sreg.required": "fullname,email"` to the parameters in the request

**Comment by JobDoesburg** [comment]
or maybe better even `email_verified` if that works, would need to check

File: website/users/models.py [hunk 1]
```diff
@@ -61,7 +61,12 @@ def __str__(self):
 
         :return: the username of the user
         """
-        return self.username
+        if self.first_name and self.last_name:
+            return "{} {}".format(self.first_name, self.last_name)
+        elif self.first_name:
+            return self.first_name
+        else:
+            return self.username
```
Comment (JobDoesburg) [inline website/users/models.py:69]: Django's built-in get_full_name() works for this right?

File: website/users/views.py [hunk 2]
```diff
@@ -67,6 +61,26 @@ class VerifyView(TemplateView):
 
     template_name = "users/verify.html"
 
+    def set_user_details(self, openid_verifier, user):
```
Comment (JobDoesburg) [inline website/users/views.py:64]: Let's move this to services

File: website/users/forms.py [hunk 3]
```diff
@@ -17,4 +17,23 @@ def clean_username(self):
...
+        help_text="This is your username used to log you in to this website.",
+        widget=forms.TextInput(attrs={"readonly": "readonly"}),
+    )
+    first_name = forms.CharField(
+        label="First name", widget=forms.TextInput(attrs={"readonly": "readonly"}),
+    )
+    last_name = forms.CharField(
+        label="Last name", widget=forms.TextInput(attrs={"readonly": "readonly"}),
+    )
+    email = forms.CharField(
```
Comment (JobDoesburg) [inline website/users/forms.py:37]: I think there is an email field for this? At least we should add an email validator

File: website/users/views.py [hunk 4]
```diff
@@ -77,16 +91,15 @@ def get(self, request, **kwargs):
...
-            except User.DoesNotExist:
-                user = User.objects.create(username=username)
-            login(request, user)
-            return redirect("index")
+        openid_verifier = get_openid_verifier(request)
+        if openid_verifier.verify_request():
+            username = openid_verifier.extract_username()
+            if username:
+                user, created = User.objects.get_or_create(username=username)
+                if created:
```
Comment (JobDoesburg) [inline website/users/views.py:99]: Maybe move this to services too

File: website/users/forms.py (hunk 3 above)
Comment (KiOui) [reply website/users/forms.py:37]: The user details can not be changed, it is only for displaying.

**Pull request merged**